page/effects/queue-and-dequeue-explained.md

```diff
@@ -4,53 +4,135 @@
+To add multiple functions to the queue, you can call `.queue()` multiple times.
 
 ```
-$.fn.pause = function( delay ) {
-	return this.queue(function() {
-		var elem = this;
-		setTimeout(function() {
-			return $( elem ).dequeue();
-		}, delay );
-	});
-};
-
 $( ".box" )
-	.animate({
+	.animate( {
 		height: 20
 	}, "slow" )
-	.pause( 1000 )
-	.animate({
-		height: 150
-	});
+	.queue( function() {
+		console.log('I fired!');
+	} )
+	.animate( {
+		height: 50
+	}, "fast" )
+	.queue( function() {
+		console.log('I fired too!');
+	} );
+```
+
+If you ran this example, you will have seen that the last animation never runs and the last callback doesn't fire. This is because we basically never told jQuery to continue. Inside of the first queued function, you will need to call `.dequeue()` to move forward to the next function in the queue.
+
+```
+.queue( function() {
+	console.log('I fired!');
+	$( this ).dequeue();
+} )
+```
+
+Another way of *dequeuing* is by calling the function that is passed to your callback. That function will automatically call `.dequeue()` for you.
+
+```
+.queue( function(next) {
+	console.log('I fired!');
+	next();
+} )
+```
+
+## Custom Queues
+
+Up to this point all of our animation and queue examples have been using the default queue name which is `fx`.  Elements can have multiple queues attached to them, and we can give each of these queues a different name.  We can specify a custom queue name as the first argument to the `.queue()` method.
+
+```
+$( ".box" )
+	.queue( "steps", function( next ) {
+		console.log( "Step 1" );
+		next();
+	} )
+	.queue( "steps", function( next ) {
+		console.log( "Step 2" );
+		next();
+	} )
+	.dequeue( "steps" );
 ```
 
-Remember that the first argument for `.queue()` and `.dequeue()` is `fx`, and that in all of these examples I'm not including it because jQuery sets the argument to `fx` by default — so I don't have to specify it.
+Notice that we have to call the `.dequeue()` method passing it the name of our custom queue to start the execution. Every queue except for the default, `fx`, has to be manually kicked off by calling `.dequeue()` and passing it the name of the queue.
+
+## Clearing The Queue
+
+Since queues are just a set of ordered operations, our application may have some logic in place that needs to prevent the remaining queue entries from executing. We can do this by calling the `.clearQueue()` method, which will empty the queue.
+
+```
+$( ".box" )
+	.queue( "steps", function( next ) {
+		console.log( "I fired" );
+		next();
+	} )
+  	.clearQueue( "steps" )
+	.dequeue( "steps" );
+```
+
+In this example, nothing will happen as we removed everything from the `steps` queue.
+
+Another way of clearing the queue is to call `.stop( true )`. That will stop the currently running animations and will clear the queue. 
+
+## Replacing The Queue
+
+When you pass an array of functions as second argument to `.queue()`, that array will replace the queue.
```

Comment:
We should add something about this not being a very "nice" thing to do, but there are cases to do it.  It's better to `.stop` or `.clearQueue` and then add new functions.